src/tools/validation.rs

```diff
@@ -0,0 +1,48 @@
+use super::prelude::*;
+pub(crate) use crate::blob::header::Header as BlobHeader;
+use crate::blob::index::FileIndexTrait;
+pub(crate) use crate::blob::index::{header::IndexHeader, BPTreeFileIndex, SimpleFileIndex};
+use crate::blob::FileName;
+pub(crate) use crate::record::{Header, Record};
+pub use crate::BloomConfig;
+use futures::Future;
+
+fn block_on<T, F: Future<Output = T>>(f: F) -> Result<T> {
+    Ok(tokio::runtime::Runtime::new()?.block_on(f))
+}
+
+/// Validate index file
+pub fn validate_index<K: Key + 'static>(path: &Path) -> AnyResult<()> {
+    let header = read_index_header(path)?;
+    let headers = match header.version() {
+        2 => block_on(async {
+            let index =
+                <SimpleFileIndex as FileIndexTrait<K>>::from_file(FileName::from_path(path)?, None)
+                    .await?;
+            let res = index.get_records_headers().await?;
+            AnyResult::<_>::Ok(res.0)
+        })?,
+        3..=4 => block_on(async {
```

Comment:
We will definitely forget to update this place if the version changes. I suggest adding a compile-time assertion that checks for `HEADER_VERSION` and generates a compile-time error if it differs from what is expected here. I found `static_assertion` crate that can do this check

Reply:
In fact, I think, that we don't need the facilities to check all possible index file versions. Because index can simply be regenerated if something wrong, then we just need to check for validity on the latest version. If index from some previous version, then it is error and it should be regenerated